### PR TITLE
added name parameter

### PR DESCRIPTION
**Pull request by himyouten**
##### SUMMARY
Added a name parameter.  The name would be used as the title for the API and would be used as a unique identifier.  As mentioned in the module already, there's no real way of having a unique identifier on the APIs, but you can use the name as a convention to be unique.

The change facilitates using the name as a unique identifier, only coming into play when the api_id is omitted.  It looks up all the APIs and gets the api_id of the first API with the name.  The modification also renames the API if the swagger info changes it.  As a side effect, the Swagger info.title can be omitted also.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
aws_api_gateway

##### ANSIBLE VERSION
```
ansible 2.4.0.0
  config file = None
  configured module search path = ['/root/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.6/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.6.2 (default, Sep 13 2017, 14:26:54) [GCC 4.9.2]
```

##### ADDITIONAL INFORMATION

```
Added
    "rename_response": {
        "created_date": "2017-09-25T21:27:55+00:00", 
        "id": "xxxx", 
        "name": "my-api", 
        "response_metadata": {
            "http_headers": {
                "connection": "keep-alive", 
                "content-length": "108", 
                "content-type": "application/json", 
                "date": "Mon, 25 Sep 2017 21:28:21 GMT", 
                "x-amzn-requestid": "12345"
            }, 
            "http_status_code": 200, 
            "request_id": "12345", 
            "retry_attempts": 0
        }
    }
```

### CODE REVIEW

File: lib/ansible/modules/cloud/amazon/aws_api_gateway.py [hunk 1]
```diff
@@ -324,5 +361,22 @@ def create_deployment(client, api_id=None, stage=None, description=None):
     return client.create_deployment(restApiId=api_id, stageName=stage, description=description)
 
 
+@AWSRetry.backoff(**retry_params)
+def get_rest_apis(client, limit=500):
+    return client.get_rest_apis(limit=limit)
```
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/aws_api_gateway.py:366]: Please use the paginators and return the full result set

http://boto3.readthedocs.io/en/latest/reference/services/apigateway.html#APIGateway.Paginator.GetRestApis

https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/GUIDELINES.md#api-throttling-and-pagination

File: lib/ansible/modules/cloud/amazon/aws_api_gateway.py [hunk 2]
```diff
@@ -122,6 +127,7 @@ def main():
     argument_spec = ec2_argument_spec()
     argument_spec.update(
         dict(
+            name=dict(type='str', required=False),
```
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/aws_api_gateway.py:130]: these are both defaults - you can just use `name=dict()`

File: lib/ansible/modules/cloud/amazon/aws_api_gateway.py [hunk 3]
```diff
@@ -30,6 +30,11 @@
 version_added: '2.4'
 requirements: [ boto3 ]
 options:
+  name:
+    description:
+      - The name of the API to manage.  If api_id is None, then the name will be
```
Comment (willthames) [inline lib/ansible/modules/cloud/amazon/aws_api_gateway.py:35]: api_id should be formatted as a module parameter (i.e. `I(api_id)`)

From http://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#formatting-options

**Review by willthames — CHANGES_REQUESTED** [review]

**Comment by mikedlr** [comment]
Hi;  The API name is already possible to configure from the YAML file which the API is uploaded from.  You can control it in Ansible by building the YAML file from a template.  What benefits do you see from having a separate name parameter?  

If we are having this, can you make it clear in the documentation which of the settings dominates the other.

**Pull request closed without merging**